### ISSUE TITLE
se pool step calculation 

### ISSUE THREAD

**Issue by makcedward** [NVIDIA/NeMo#3788]
https://github.com/NVIDIA/NeMo/blob/256236f684cd08abd27008d6a9c41d77993be138/nemo/collections/asr/parts/submodules/jasper.py#L174

https://github.com/NVIDIA/NeMo/blob/256236f684cd08abd27008d6a9c41d77993be138/nemo/collections/asr/parts/submodules/jasper.py#L506

It seems that no matter what timesteps and context_window are, same formula is applied. 

By the way, suggesting to use `.to` instead of `.type`. [Reference](https://discuss.pytorch.org/t/why-does-pytorch-needs-the-three-functions-to-type-and-type-as/20164/2)

**Comment by titu1994** [comment]
We experimented with quite a few options, and most were worse than what we ended up with. During inference if the chunk size is small enough then the average during inference takes into account the whole chunk. If it's larger, then there's a mismatch between train and inference normalization steps and the model predictions deteriorated. 

This is not an optimal solution, just the most robust one to train-test time difference in chunk sizes. 

We overall find that any form of limited SE is highly unstable to train, and even when it does train is much worse wrt WER at inference time (both with limited and unlimited chunk). Therefore we have paused efforts on this front. We find that the attention mechanism is highly robust to such issues and can be more easily controlled.
